### PR TITLE
 Add typing for DefinePlugin

### PR DESCRIPTION
**Pull request by mohsen1**
**What kind of change does this PR introduce?**
typing

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
no

Waiting for Compiler types

### CODE REVIEW

**Comment by mohsen1** [comment]
@sokra @TheLarkInn this is good to go

File: lib/DefinePlugin.js [hunk 1]
```diff
@@ -42,10 +52,19 @@ const toCode = code => {
...
+	 * Create a new define plugin
+	 * @param {Object} definitions A map of global object definitions
+	 */
 	constructor(definitions) {
 		this.definitions = definitions;
 	}
 
+	/**
+	 * Apply the plugin
+	 * @param {TODO} compiler Webpack Compiler
```
Comment (sokra) [inline lib/DefinePlugin.js]: TODO -> Compiler

File: lib/DefinePlugin.js [hunk 2]
```diff
@@ -57,7 +76,18 @@ class DefinePlugin {
 					new ConstDependency.Template()
 				);
 
+				/**
+				 * Handler
+				 * @param {TODO} parser Parser
```
Comment (sokra) [inline lib/DefinePlugin.js]: TODO -> Parser

File: lib/DefinePlugin.js [hunk 3]
```diff
@@ -57,7 +76,18 @@ class DefinePlugin {
...
+				 * @returns {void}
+				 */
 				const handler = parser => {
+					/**
+					 * Walk definitions
+					 * @param {TODO} definitions Definitions map
```
Comment (sokra) [inline lib/DefinePlugin.js]: TODO -> Object

File: lib/DefinePlugin.js [hunk 4]
```diff
@@ -22,6 +27,11 @@ const stringifyObj = obj => {
 	);
 };
 
+/**
+ * Convert code to a string that evaluates
+ * @param {null|undefined|RegExp|Function|Object} code Code to evaluate
```
Comment (sokra) [inline lib/DefinePlugin.js]: Could you add a typedef for this? i. e. `CodeValue`

File: lib/DefinePlugin.js [hunk 5]
```diff
@@ -85,6 +121,12 @@ class DefinePlugin {
 						});
 					};
 
+					/**
+					 * Apply Code
+					 * @param {string} key Key
+					 * @param {TODO} code Code
```
Comment (sokra) [inline lib/DefinePlugin.js]: TODO -> CodeValue

**Comment by mohsen1** [comment]
@sokra this should be good to go

**Pull request closed without merging**